lib/ecto/query/builder/order_by.ex

```diff
@@ -50,9 +50,12 @@ defmodule Ecto.Query.Builder.OrderBy do
   @spec escape(:order_by | :distinct, Macro.t(), {list, term}, Keyword.t(), Macro.Env.t()) ::
           {Macro.t(), {list, term}}
   def escape(kind, expr, params_acc, vars, env) do
-    expr
-    |> List.wrap()
-    |> Enum.map_reduce(params_acc, &do_escape(&1, &2, kind, vars, env))
+    {ast, params_acc} =
+      expr
+      |> List.wrap()
+      |> Enum.map_reduce(params_acc, &do_escape(&1, &2, kind, vars, env))
+
+    {List.flatten(ast), params_acc}
```

Comment:
I don't think this is the correct approach. We don't want to allow someone to write: `order_by: [[asc: p.created_at]]` and I believe this change would allow it.

Reply:
You are right, changed.